lib/reek/smell_detectors/prima_donna_method.rb

```diff
@@ -28,9 +28,27 @@ def self.contexts # :nodoc:
         [:class]
       end
 
+      #
+      # @param ctx [Context::ModuleContext]
+      # @return [Array<SmellWarning>]
+      #
+      # Given this code:
+      #
+      # class Alfa
+      #   def bravo!
+      #   end
+      # end
+      #
+      # An example `ctx` could look like this:
+      #
+      # s(:class,
+      #   s(:const, nil, :Alfa), nil,
+      #     s(:def, :bravo!,
+      #       s(:args), nil))
+      #
       def sniff(ctx)
         ctx.node_instance_methods.select do |method_sexp|
-          prima_donna_method?(method_sexp, ctx)
+          prima_donna_method?(method_sexp, ctx) && !ignore_method?(method_sexp, ctx)
```

Comment:
Actually, the best option may be to have the `ignore_method?` check between `method_sexp.ends_with_bang?` and `version_without_bang_exists?(method_sexp, ctx)`.